apps/root/src/common/utils/transaction-history/index.ts

```diff
@@ -186,3 +186,34 @@ export const getTransactionPriceColor = (tx: TransactionEvent) => {
 
   return undefined;
 };
+
+export const getTransactionInvolvedWallets = (tx: TransactionEvent) => {
+  let wallets: string[] = [];
+  switch (tx.type) {
+    case TransactionEventTypes.ERC20_APPROVAL:
+      const { owner, spender } = tx.data;
+      wallets = [owner, spender];
+      break;
+    case TransactionEventTypes.DCA_WITHDRAW:
+    case TransactionEventTypes.DCA_TERMINATED:
+    case TransactionEventTypes.DCA_MODIFIED:
+    case TransactionEventTypes.DCA_PERMISSIONS_MODIFIED:
+    case TransactionEventTypes.DCA_CREATED:
```

Comment:
DCA_CREATED could have a different user to the one that initiated the tx

Reply:
Not through our UI, but it is available through SC so an event could come with those two values different

Reply:
Pretty cool!